orchestrator/relayer/src/ibc_auto_forwarding.rs

```diff
@@ -0,0 +1,91 @@
+/// Note that this is necessary due to a Tendermint bug preventing Gravity from initiating IBC transfers
+/// in EndBlocker. Moving the IBC transfers to a queue which can be cleared in a Tx solves the issue.
+#[allow(clippy::too_many_arguments)]
+pub async fn ibc_auto_forward_loop(
+    cosmos_key: Option<CosmosPrivateKey>,
+    contact: &Contact,
+    grpc_client: GravityQueryClient<Channel>,
+    fee: Option<Coin>,
+    relayer_config: RelayerConfig,
+) {
+    let mut grpc_client = grpc_client;
+
+    if cosmos_key.is_none() {
+        error!("Unable to execute pending ibc auto forwards with no configured private key!");
+        return;
+    }
+    let cosmos_key = cosmos_key.unwrap();
+
+    let fee = match fee {
+        None => Coin { amount: 0u8.into(), denom: "ugraviton".to_string() },
+        Some(f) => f,
+    };
+
+    loop {
+        let loop_start = Instant::now();
+        let pending_forwards = get_pending_ibc_auto_forwards(&mut grpc_client).await;
+        let should_execute_pending_ibc_auto_forwards = pending_forwards.len() > 0;
+
+        if should_execute_pending_ibc_auto_forwards {
+            info!("Executing {}/{} pending ibc auto forwards", relayer_config.ibc_auto_forwards_to_execute, pending_forwards.len());
+
+            let res = execute_pending_ibc_auto_forwards(contact, cosmos_key, fee.clone(), relayer_config.ibc_auto_forwards_to_execute).await;
+            if res.is_err() {
+                warn!("Error submitting MsgExecuteIbcAutoForwards! {}", res.err().unwrap())
+            }
+        }
+
+        // a bit of logic that tries to keep things running every ibc_auto_forward_loop_speed seconds
+        // exactly, this is not required for any specific reason.
+        let elapsed = Instant::now() - loop_start;
+        let loop_speed = Duration::from_secs(relayer_config.ibc_auto_forward_loop_speed);
+        if elapsed < loop_speed {
+            delay_for(loop_speed - elapsed).await;
+        }
+    }
+}
+
+/// Queries the Gravity chain for Pending Ibc Auto Forwards, returning an empty vec if there is an error
+pub async fn get_pending_ibc_auto_forwards(grpc_client: &mut GravityQueryClient<Channel>) -> Vec<PendingIbcAutoForward> {
+    let pending_forwards = grpc_client.get_pending_ibc_auto_forwards(QueryPendingIbcAutoForwards{limit: 0}).await;
+    if pending_forwards.is_err() {
+        let status = pending_forwards.err().unwrap();
+        warn!("Received an error when querying for pending ibc auto forwards: {}", status.message());
+        return vec![];
+    }
+
+    let pending_forwards = pending_forwards.unwrap();
+    return pending_forwards.into_inner().pending_ibc_auto_forwards;
+}
+
+/// Executes a MsgExecuteIbcAutoForwards on the gravity chain, which will process forwards_to_clear number of pending ibc auto forwards
+pub async fn execute_pending_ibc_auto_forwards(contact: &Contact, cosmos_key: CosmosPrivateKey, fee: Coin, forwards_to_clear: u64) -> Result<(),CosmosGrpcError> {
```

Comment:
this should probably be a cosmos-gravity utility function, we'll want to be able to easily call it in the test_runner solo at some point.